server/routerlicious/packages/services-shared/src/runnerUtils.ts

```diff
@@ -8,6 +8,46 @@ import { IWebServer } from "@fluidframework/server-services-core";
 import { Lumber, Lumberjack } from "@fluidframework/server-services-telemetry";
 import { promiseTimeout } from "@fluidframework/server-services-client";
 import { Deferred } from "@fluidframework/common-utils";
+import type { IRedisClientConnectionManager } from "@fluidframework/server-services-utils";
+
+/**
+ * @internal
+ */
+export async function closeRedisClientConnections(
+	redisClientConnectionManagers: IRedisClientConnectionManager[],
+): Promise<void> {
+	const redisClientsQuittingP: Promise<void | string>[] = [];
+	for (const redisClientConnectionManager of redisClientConnectionManagers) {
+		try {
+			const redisClient = redisClientConnectionManager.getRedisClient();
+			redisClientsQuittingP.push(
+				redisClient
+					.quit()
+					.then(() => {
+						Lumberjack.info("Redis client quit");
+						return "OK";
+					})
+					.catch((error) => {
+						if (error?.message === "Connection is closed.") {
+							// Ignore the error caused by disconnecting since
+							// we're disconnecting...
+							// https://github.com/redis/ioredis/blob/v4/lib/cluster/index.ts#L415
+							Lumberjack.info("Redis client already closed");
+							return "OK";
+						}
+						Lumberjack.error("Failed to quit redis client", undefined, error);
+					}),
+			);
+		} catch (error) {
+			if (error instanceof Error && error.message === "Redis client not initialized") {
```

Comment:
This seems unnecessary. `error` will be logged twice, including its message, because this falls through, meaning this condition just adds a third `error.message` log. There doesn't seem to be a benefit to adding this condition here.